### DISCUSSION TITLE
Basic question on encryption

### DISCUSSION THREAD

**Discussion by paulsommer** [rustdesk/rustdesk#4478] [General]
Hi,

i just changed from Anydesk to Rustdesk and are quite impressed of it's functionality (especial V1.2.0) and speed. Many thanks for that great work.

I have setup my own ID- and relay server successfully but unfortunately I still don't understand the encryption technique.

1. What exactly is encrypted (or not encrypted)? The connection from the client to the hbbr/hbbs server or the connection between the two clients, or both?

2. When I enter the servers public key in all of my clients, everything goes well.
When I enter no (or a wrong) key in the client everything goes well, too! There is no warning message that this is an unencrypted connection.
So, is this connection encrypted or not, **and how to find out?**

3. When I use the  -k_ parameter tn the rustdesk service file, then the key is mandatory for a successful connection. 
But how can a person that has just downloaded and installed the software enter that lengthy key to establish the first contact?

4. Where are the keys of the clients are stored (Windows and Linux)?

Why don't the client and the server send their public keys automatically?

Please enlighten me :-)

**Comment by SergeyMy** [comment]
>        So, is this connection encrypted or not, **and how to find out?**
>When I don't enter a (or wrong) key in the client, everything is unencrypted
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/4c4c8124-fff3-46d5-90f1-14e5d20915a3)
>When I enter the servers public key in all of my clients, everything goes well.
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/2a4add6f-1e3b-42f0-9be0-82f0db176e39)

>    What exactly is encrypted (or not encrypted)? 
https://github.com/rustdesk/rustdesk/wiki/How-does-RustDesk-work%3F

> Where are the keys of the clients are stored 
~/.config/rustdesk/RustDesk2.toml

**Comment by paulsommer** [comment]
Oh, cool. I haven't noticed that little icon.

I just found out, that only the supporter needs the key to be set to get a encrypted connection.

Very strange to me. Is that an inteded behaviour?

**Comment by paulsommer** [comment]
This case is still open.

Sometimes I get an unexpected Message: "Keys do not match", even with clients that used to work before. When I examine the settings the Key-field is empty. After setting the key it's working again.

That raises two questions:
1. Why does it work sometimes when no key is set or even a wrong key is set at client side?
2. How to support a new client? Nobody can enter that key manually, and sending it with email is not a good idea, I think. (and it's too complicated for noobs anyway)
